stan/inc2prev.stan

```diff
@@ -50,95 +53,108 @@ data {
+  vector<lower = 0, upper = 1>[n_ab > 0 ? 1 : 0] delta; // vaccine efficacy
+  vector<lower = 0, upper = 1>[n] init_dab; // initial proportion with antibodies
+  vector<lower = 0>[n_ab > 0 ? 1 : 0] k; // Potential loss of efficacy from new infections in already seropositive people
+  vector<lower = 0>[n_ab > 0 ? 1 : 0] l; // Potential loss of efficacy from new doses being administered to already seropositive people
 }
 
 transformed parameters {
-  vector[t] gp; // value of gp at time t + initialisation 
-  vector[t] infections; // incident infections at time t
-  vector[ab_obs ? t : 0] infs_with_potential_abs; // Infections with the potential to have ab
-  vector<lower = 0, upper = 1>[t] dcases; // detectable cases at time t
-  vector[ab_obs ? t : 0] dab; // proportion of individuals with antibodies at time t
-  vector[obs] odcases;
-  vector[ab_obs] odab;
-  vector[obs] combined_sigma;
-  vector[ab_obs] combined_ab_sigma;
+  matrix[n, t] gp; // value of gp at time t + initialisation
+  matrix[n, t] infections; // incident infections at time t
+  matrix[n_ab, t] infs_with_potential_abs; // Infections with the potential to have ab
+  matrix[n, t] dcases; // detectable cases at time t
+  matrix[n_ab, t] dab; // proportion of individuals with antibodies at time t
+  matrix[n, obs] odcases;
+  matrix[n_ab, ab_obs] odab;
+  matrix[n, obs] combined_sigma;
+  matrix[n_ab, ab_obs] combined_ab_sigma;
   // update gaussian process
-  gp[(1 + diff_order):t] = update_gp(PHI, M, L, alpha, rho, eta, 0);
-  // setup differencing of the GP
-  if (diff_order) {
-    gp[1:diff_order] = init_growth;
-    for (i in 1:diff_order) {
-      gp = cumulative_sum(gp);
+  for (i in 1:n) {
+    gp[i, (1 + diff_order):t] = to_row_vector(update_gp(PHI, M, L, alpha[i], rho[i], to_vector(eta[i]), 0));
+    // setup differencing of the GP
+    if (diff_order) {
+      gp[i, 1:diff_order] = to_row_vector(init_growth[i]);
+      for (j in 1:diff_order) {
+        gp[i] = cumulative_sum(gp[i]);
+      }
     }
-  }
 
-  // relative probability of infection
-  // inc_init is the mean incidence
-  infections = inv_logit(init_inc + gp);
+    // relative probability of infection
+    // inc_init is the mean incidence
+    infections[i] = inv_logit(init_inc[i] + gp[i]);
 
-  // calculate detectable cases
-  dcases = convolve(infections, prob_detect);
-  // calculate observed detectable cases
-  odcases = observed_in_window(dcases, prev_stime, prev_etime, ut, obs);
-  //combined standard error
-  combined_sigma = sqrt(square(sigma) + prev_sd2);
+    // calculate detectable cases
+    dcases[i] = to_row_vector(convolve(to_vector(infections[i]), prob_detect));
+    // calculate observed detectable cases
+    odcases[i] = to_row_vector(observed_in_window(to_vector(dcases[i]), prev_stime, prev_etime, ut, obs));
+    //combined standard error
+    combined_sigma[i] = sqrt(rep_row_vector(square(sigma), obs) + prev_sd2[i]);
```

Comment:
so everybody shares the same observation error? I feel like this shouldn't need a rep statement?